src/Hamiltonians/ExtendedHubbardReal1D.jl

```diff
@@ -13,20 +13,45 @@ Implements the extended Hubbard model on a one-dimensional chain in real space.
 * `u`: on-site interaction parameter
 * `v`: the next-neighbor interaction
 * `t`: the hopping strength
+* `boundary_condition` : applied boundray condition 
+
+# Boundary conditions
+### For real Hamiltonian
+* For π-twisted -> :twisted
+* For hard wall -> :hard_wall
+### For complex Hamiltonian
+* For θ-twisted -> Number "θ"
+
+where,
+
+```math
+a_{M+1} = exp(-\\iota θ) a_1
+```
 
 """
-struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T} <: AbstractHamiltonian{TT}
+struct ExtendedHubbardReal1D{TT,A<:SingleComponentFockAddress,U,V,T,BOUNDARY_CONDITION} <: AbstractHamiltonian{TT}
     add::A
 end
 
 # addr for compatibility.
-function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0)
-    U, V, T = promote(float(u), float(v), float(t))
-    return ExtendedHubbardReal1D{typeof(U),typeof(addr),U,V,T}(addr)
+function ExtendedHubbardReal1D(addr; u=1.0, v=1.0, t=1.0, boundary_condition = :periodic)
+    if boundary_condition == :periodic || boundary_condition == :twisted || boundary_condition == :hard_wall
+        U, V, T = promote(float(u), float(v), float(t))
+        return ExtendedHubbardReal1D{typeof(U),typeof(addr),U,V,T,boundary_condition}(addr)
+    elseif boundary_condition isa Number
+        U, V, T = promote(complex(u), complex(v), complex(t))
+        return ExtendedHubbardReal1D{typeof(U),typeof(addr),U,V,T,boundary_condition}(addr)
```

Comment:
So the problem with this approach is that all parameters end up as complex numbers, which they don't need to be if real numbers are passed.
As a further consequence `LOStructure` will default to `AdjointUnknown()`, because everything is complex, while indeed it should be `IsHermitian()` as long as `u` and `v` are real numbers.

I would suggest to write this as
```suggestion
        U, V, T = promote(float(u), float(v), float(t))
        return ExtendedHubbardReal1D{typeof(complex(U)),typeof(addr),U,V,T,boundary_condition}(addr)
```
This is not yet enough to deal with `LOStructure`. We need to add a new method to `LOStructure` to make sure it recognizes this Hamiltonian as hermitian. We need another method (in addition to the one defined in line 62) that marks `ExtendedHubbardReal1D` Hamiltonian with real-valued parameters but number-twisted boundary conditions as `IsHermitian()`.

You may also need to add explicit type casts to `eltype(h)` when returning values from `diagonal_element` and `get_offdiagonal`. See below.

```